File: Python/2610-arrayTo2D.py
from typing import List
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        lists = [[]]
        added = False; 
        for num in nums:
            for list in lists:
                if (not num in list and not added):
                    list.append(num)
                    added = True
            if (not added):
                list = [num]
                lists.append(list)
            added = False
        return lists

mi_instancia = Solution()
mi_lista_de_numeros = [1,3,4,1,2,3,1]
resultado = mi_instancia.findMatrix(mi_lista_de_numeros)

File: TypeScript/2610-arrayTo2D.ts
function findMatrix(nums: number[]): number[][] {
    let matrix: number[][]=[];
    let found: boolean = false; 
    let added: boolean = false; 
    let newRow: number[] = [];
    matrix.push(newRow);
    for (let index = 0; index < nums.length; index++) {
        added =false;
        for (let row = 0; row < matrix.length; row++) {
            // for (let column = 0; column < matrix[0].length; column++) {
            let column =0;
            found = false;
            while (column < matrix[row].length && !found){    
                if (matrix[row][column] === nums[index]){
                    found = true;
                }
                column++;
            }
            if (!found && !added){
                matrix[row].push(nums[index]);
                added =true;
            }            
        }
        if (!added){
            let newRow: number[] = [];
            newRow.push(nums[index]);
            matrix.push(newRow);
        }        
    }
    return matrix;
}